fsspec/implementations/tests/test_smb.py

```diff
@@ -0,0 +1,94 @@
+import shlex
+import subprocess
+import time
+import pytest
+import fsspec
+
+pytest.importorskip("smbprotocol")
+
+
+def stop_docker(container):
+    cmd = shlex.split('docker ps -a -q --filter "name=%s"' % container)
+    cid = subprocess.check_output(cmd).strip().decode()
+    if cid:
+        subprocess.call(["docker", "rm", "-f", "-v", cid])
+
+
+@pytest.fixture(scope="module")
+def smb_params():
+    try:
+        pchk = ["docker", "run", "--name", "fsspec_test_smb", "hello-world"]
+        subprocess.check_call(pchk)
+    except (subprocess.CalledProcessError, FileNotFoundError):
+        pytest.skip("docker run not available")
+        return
+    stop_docker("fsspec_test_smb")
+
+    # requires docker
+    container = "fsspec_smb"
+    stop_docker(container)
+    img = "docker run --name {} --detach -p 139:139 -p 445:445 dperson/samba"
+    cfg = " -p -u 'testuser;testpass' -s 'home;/share;no;no;no;testuser'"
+    cmd = img.format(container) + cfg
+    cid = subprocess.check_output(shlex.split(cmd)).strip().decode()
+    print(cid)
+    try:
+        time.sleep(1)
+        yield dict(host="localhost", port=445, username="testuser", password="testpass")
+    finally:
+        import smbclient
+
+        smbclient.reset_connection_cache()
+        stop_docker(container)
+
+
+def test_simple(smb_params):
+    adir = "/home/adir"
+    adir2 = "/home/adir/otherdir/"
+    afile = "/home/adir/otherdir/afile"
```

Comment:
Please forget my earlier comment, you clearly show that ordinary paths work as expected, *and* that whole URLs are accepted by `open`, just as it should be.